File: src/my_project/easy_problems/from151To200/two_out_of_three.py
from typing import List 

class Solution:
    def twoOutOfThree(self, nums1: List[int], nums2: List[int], nums3: List[int]) -> List[int]:

        temp = list()
        for num in nums1:
            if num not in temp:
                temp.append(num)
        nums1 = temp

        temp = []
        for num in nums2:
            if num not in temp:
                temp.append(num)
        nums2 = temp

        temp = []
        for num in nums3:
            if num not in temp:
                temp.append(num)
        nums3 = temp

        temp_dict = {}

        for num in nums1 + nums2 + nums3:
            temp_dict[num] = 0
        
        for num in nums1 + nums2 + nums3:
            temp_dict[num] += 1

        answer = list()

        for num in temp_dict.keys():
            if temp_dict[num] >= 2:
                answer.append(num)



        return answer

solution = Solution()

print(solution.twoOutOfThree(nums1 = [1,1,3,2], nums2 = [2,3], nums3 = [3]))

print(solution.twoOutOfThree(nums1 = [3,1], nums2 = [2,3], nums3 = [1,2]))

print(solution.twoOutOfThree(nums1 = [1,2,2], nums2 = [4,3,3], nums3 = [5]))

File: src/my_project_typescript/easy_problems_typescript/from51To100/twoOutOfThree.ts
function twoOutOfThree(nums1: number[], nums2: number[], nums3: number[]): number[] {

    let temp: Array<number> = []


    nums1.map( num => {
        if (!temp.includes(num)){
            temp.push(num)
        }
    })
    nums1 = temp 

    temp = []
    nums2.map( num => {
        if (!temp.includes(num)){
            temp.push(num)
        }
    })
    nums2 = temp 

    temp = []
    nums3.map( num => {
        if (!temp.includes(num)){
            temp.push(num)
        }
    })
    nums3 = temp 

    let tempDict: any = {}

    for (let num of nums1.concat(nums2).concat(nums3)){
        tempDict[num] = 0
    }

    for (let num of nums1.concat(nums2).concat(nums3)){
        tempDict[num] +=1 
    }

    let answer: Array<number> =[]
    for (let key in tempDict){
        if (tempDict[key] >= 2){
            answer.push(parseInt(key))
        }
    }




    return answer
};

console.log(twoOutOfThree([1,1,3,2],  [2,3], [3]))

console.log(twoOutOfThree([3,1],  [2,3],  [1,2]))

console.log(twoOutOfThree([1,2,2],  [4,3,3], [5]))
